Fully document External Data (#73)
pomerium/documentation: content/docs/enterprise/external-data/bamboohr.mdx

---
title: BambooHR
description: Configure the BambooHR external data provider to extend your access policies.
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

The BambooHR integration provides integration with HR data such as group memberships, employment status, out of the office, location, etc.

:::caution

> The external data sources we provide are meant to be examples and inspiration for users to create their own data sources. We won't maintain these integrations in perpituity, and changes to the APIs they interact with may break them in the future.

Reviewer: ```suggestion
The external data sources we provide are meant to be examples and inspiration for users to create their own data sources. We won't maintain these integrations in perpetuity, and changes to the APIs they interact with may break them in the future.
```

The [datasource](https://github.com/pomerium/datasource) project is open-source, and if the community wishes to provide contributions to keep it working in the future, we will shepherd those updates.
:::

## Install

1. Create new BambooHR API key by navigating to **Your profile** → **API Keys**, and adding a new API Key. 

> 1. In order to correctly parse dates returned by BambooHR API you will need provide a time zone - either an `UTC` or [IANA Time Zone] database name, i.e. `America/New_York`. 

Reviewer: ```suggestion
1. In order to correctly parse dates returned by BambooHR API you will need to provide a time zone - either an `UTC` or [IANA Time Zone] database name, i.e. `America/New_York`. 
```

  cd pomerium-datasource-*
  ./pomerium-datasource bamboohr --bamboohr-api-key=$YOUR_API_KEY --bamboohr-subdomian=$YOUR_BAMBOOHR_SUBDOMAIN --bamboohr-time-zone=America/New_York
  ```

  The output should resemble:

  ```sh
  {"level":"info","message":"ready"}
  ```

</TabItem>
</Tabs>

## Configure Configure External Data Source

BambooHR data connector exposes two API endpoints: 

- `/employees/all` returns all employees 
- `/employees/available` returns employees that are not currently out of the office due to vacation or other leave reasons. 

To create new external data record:

1. In the Pomerium Enterprise Console, navigate to **CONFIGURE** → **External Data** and click **+ ADD EXTERNAL DATA SOURCE**.

1. Fill out the following fields:

  | Field       | Content                                    | Notes                                                                             |
  | :---------- | :----------------------------------------- | :-------------------------------------------------------------------------------- |
  | URL         | `http://bamboohr:8080/employees/available` | Adjust for the endpoint you'll write policies against.                            |

>   | Record type | `pomerium.io/BambooHRAvailable`            | As above, adjust to somethinglike pomerium.io/BambooHRAll for the other endpoint. |

Reviewer: ```suggestion
  | Record type | `pomerium.io/BambooHRAvailable`            | As above, adjust to something like pomerium.io/BambooHRAll for the other endpoint. |
```
